app/farm/views.py

```diff
@@ -0,0 +1,80 @@
+from django.http import HttpResponseRedirect
+from django.shortcuts import render, get_object_or_404
+from django.urls import reverse
+
+from .form import FarmDetailsForm
+from .models import Farm
+
+
+# Landing page that display the list of farms that belongs to the currently logged in user
+def index(request):
+    farm_list = Farm.objects.filter(farmer=request.user.id).order_by('-name')[:5]
+    context = {'farm_list': farm_list}
+    return render(request, 'farm/index.html', context)
+
+
+# Handler to display farm details
+def details(request, farm_id):
+    try:
+        # filter by user id to prevent information leakage
+        farm = Farm.objects.filter(pk=farm_id).filter(farmer=request.user.id).get()
+    except Farm.DoesNotExist:
+        return render(request, 'farm/details_form.html', {'error_message': 'Invalid farm ID'})
+
+    else:
+        return render(request, 'farm/details_form.html',
+                      {'form': FarmDetailsForm(instance=farm, use_required_attribute=True)})
+
+
+# Handler for adding a new farm
+def add(request):
+    if request.method == 'POST':
+        form = FarmDetailsForm(request.POST)
+        if form.is_valid():
+            farm = Farm(
+                name=form.cleaned_data['name'],
+                description=form.cleaned_data['description'],
+                phone_no=form.cleaned_data['phone_no'],
+                farmer=request.user
+            )
+            farm.save()
```

Comment:
This could be made more succinct by using the spread operator:
```suggestion
        if form.is_valid():
            farm = Farm(**form.cleaned_data, farmer=request.user)
            farm.save()
```

Reply:
Thanks for the tip. Done.